libs/vscode/nx-project-view/src/lib/views/nx-project-util.ts

```diff
@@ -0,0 +1,23 @@
+import path = require('node:path');
+
+export function isDefined<T>(val?: T): val is T {
+  return !!val;
+}
+
+export const PathHelper = {
+  dirs(val: string) {
+    return val.split(path.sep);
```

Comment:
As discussed, manually splitting is prone to errors. Try to use node.js path functions.

Reply:
Fixed and validated via unit tests.